File: hackerrank/30-days-of-code/sorting/sorting.py
#!/usr/bin/env python

import sys

def swap(n, a):
    numSwaps = 0
    for i in range(n):
        for j in range(0, n-i-1):
            if a[j] > a[j + 1]:
                a[j], a[j + 1] = a[j + 1], a[j]
                numSwaps += 1
    return numSwaps

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))

num_of_swaps = swap(n, a)

print("Array is sorted in ", num_of_swaps, " swaps.")
print("First Element: ", a[0])
print("Last Element: ", a[-1])

File: hackerrank/30-days-of-code/sorting/sorting.cpp
#include <bits/stdc++.h>

using namespace std;

// void swap(int a, int b) {
//     int temp = a;
//     a = b;
//     b = temp;
// }

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for(int a_i = 0; a_i < n; a_i++){
    	cin >> a[a_i];
    }
    // Write Your Code Here
    int numSwaps = 0;
    for(int i = 0;i < n;i++){
        for(int j = 0;j < n-1;j++){
            if (a[j] > a[j + 1]) {
                int temp = a[j];
                a[j] = a[j + 1];
                a[j + 1] = temp;
                // swap(a[j], a[j + 1]);
                numSwaps++;
            }
        }
        for (int i = 0; i < n; i++)
        {
            cout << a[i] << ' ';
        }
        cout << endl;
    }
    cout << "Array is sorted in " << numSwaps << " swaps." << endl;
    cout << "First Element: " << a[0] << endl;
    cout << "Last Element: " << a[-1] << endl;
    return 0;
}

